### ISSUE TITLE
add Lab screen

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/equality-explorer-basics#3]
Add a screen that allows the user to change the values of the mystery objects. 

@amanda-phet please specify:

- [x] name for the screen (tentatively "Lab")
**AM: Lab fits the paradigm we use in other sims so I think that is fine here.**
- [x] icon(s) for home screen and navigation bar
**square = picker**
- [x] the 3 objects, and associated artwork (icon and shadow) if they are new. (Please provide .ai file and credits.)
**shapes: sphere, square, triangle**
- [x] default values, and value ranges
**AM: defaults should all be 1. Ranges can be [1,10].**
- [x] control for changing the values (tentatively 3 pickers arranged horizontally in an accordion box below the snapshots accordion box)
**AM: 3 pickers arranged horizontally are working well so I'd like to keep that.**
- [x] title for the accordion box
**AM: Values**

**Comment by pixelzoom** [comment]
@amanda-phet What are we going to do about mystery object weights in the Snapshots dialog?  In the "Variables" screen,  we save the value of 'x' with the snapshot, and allow the user to show the value of x in the snapshot via a checkbox.  (See screenshot below.)   We don't have room to do something like this for 3 mystery objects, where the snapshot equation are already quite small. (See 2nd screenshot below.)

![screenshot_602](https://user-images.githubusercontent.com/3046552/39606464-d358aec2-4ef2-11e8-8b7b-70a8156dc796.png)

![screenshot_603](https://user-images.githubusercontent.com/3046552/39606501-086edd3e-4ef3-11e8-90fe-8f469271d978.png)

**Comment by pixelzoom** [comment]
@amanda-phet I did a sprint on this screen, see [1.0.0-dev.2](https://phet-dev.colorado.edu/html/equality-explorer-basics/1.0.0-dev.2/phet/equality-explorer-basics_en_phet.html).  Things came together very quickly; this is ~2 hours of work. 

For the objects, I used fruits, but you can specify other objects.  I labeled the accordion box "Weights" and displayed the title while the accordion box is open.  I reduced the number of snapshots to 4. (All of this is easy to change.)

Three problems, all related to Snapshots:

(1) Snapshots are not capable of saving weights for mystery objects, or saving values for multiple "variables".  If we want to save weights (which I'm guessing we will), I'll need to (1) convert mystery terms to a type of variable term,  and (2) modify Snapshots to save multiple variables. 2-6 hours.

(2) Snapshots don't have enough room to display multiple "weight" values, see https://github.com/phetsims/equality-explorer-basics/issues/3#issuecomment-386461432.  This is a design layout problem; we don't have enough horizontal space.  My only thought on how to solve it is to reduce the number of mystery objects from 3 to 2. 

(3) If we want to display "weight" values with the Snapshots, we'll probably want to show/hide values with a checkbox, ala the "x = ?" checkbox on the Variables screen.   What would the label on that checkbox be in this case?

**Comment by amanda-phet** [comment]
>Snapshots are not capable of saving weights for mystery objects, or saving values for multiple "variables". If we want to save weights (which I'm guessing we will), I'll need to (1) convert mystery terms to a type of variable term, and (2) modify Snapshots to save multiple variables. 2-6 hours.

These definitely need to be saved. An alternative would be simply not having snapshots... I should think about whether we really need snapshots here or not. 

>Snapshots don't have enough room to display multiple "weight" values, see #3 (comment). This is a design layout problem; we don't have enough horizontal space. My only thought on how to solve it is to reduce the number of mystery objects from 3 to 2.

I want to keep 3 mystery objects. If we want to display the values we'd need to find a new way to do that. I'm starting to think we might not want snapshots anymore because the current snapshots design won't be usable. 

>If we want to display "weight" values with the Snapshots, we'll probably want to show/hide values with a checkbox, ala the "x = ?" checkbox on the Variables screen. What would the label on that checkbox be in this case?

Let's address this when I figure out the items above.

**Comment by amanda-phet** [comment]
@pixelzoom I added inline responses to the original checklist and will continue to do that to address those items.

**Comment by kathy-phet** [comment]
Fun to play with this! Thanks for the quick mock up. 

> Snapshots don't have enough room to display multiple "weight" values, see #3 (comment). This is a design layout problem; we don't have enough horizontal space. My only thought on how to solve it is to reduce the number of mystery objects from 3 to 2.


> I want to keep 3 mystery objects. If we want to display the values we'd need to find a new way to do that. I'm starting to think we might not want snapshots anymore because the current snapshots design won't be usable.

I was thinking something like this could work regarding snapshots and values display. I like the snapshots for the power of collecting the system of equations that govern a particular set of values.   
![image](https://user-images.githubusercontent.com/5096804/39651107-da99f5f0-4fa7-11e8-9aa5-2ec9c26c72b5.png)

**Comment by pixelzoom** [comment]
What @kathy-phet has mocked up above (equation and weights on separate lines) is visually pleasing.  But two notes: (1) we'd have room for 4 snapshots max, and (2) this would be an entirely new implementation of the snapshot equations, and involves modifying the existing implementation.

**Comment by pixelzoom** [comment]
I've addressed problem (1) described in https://github.com/phetsims/equality-explorer-basics/issues/3#issuecomment-386472075.  Snapshots are now able to save/restore the weights of mystery objects.  I devised a simple way to do this that took 30 minutes and minimal changes to implement.

**Comment by amanda-phet** [comment]
@kathy-phet it's like you read my mind! That's exactly what I was picturing, but since we'd need to restructure the objects to behave like variables, AND we'd need to redesign the snapshots for this screen only, I wanted to discuss the idea with @pixelzoom to see how long it might take.

**Comment by pixelzoom** [comment]
Here's what I need to do to make this happen:

- [x] make mystery terms a type of variable term (a relatively big change), see https://github.com/phetsims/equality-explorer/issues/105
- [x] Add multi-line support to Snapshots, where variable values are on a separate line
- [x] Configure SnapshotControl with `orientation:'vertical'`, `commaSeparated:false`
- [x] Show multiple pickers in Variable accordion box
- [x] Provide option to change title of VariablesAccordionBox
- [x] use VariablesAccordionBox in Lab screen

2-3 days of work, affects both sims, needs to be done before dev testing.

**Comment by pixelzoom** [comment]
The work that I identified in https://github.com/phetsims/equality-explorer-basics/issues/3#issuecomment-387452611 is done, see [1.0.0-dev.3](https://phet-dev.colorado.edu/html/equality-explorer-basics/1.0.0-dev.3/phet/equality-explorer-basics_en_phet.html).

@amanda-phet I'll need the remaining items in https://github.com/phetsims/equality-explorer-basics/issues/3#issue-320104212, plus any change requests, to complete the implementation.

@kathy-phet @ariel-phet FYI.

**Comment by pixelzoom** [comment]
Perhaps something like this for the screen icon, where the apple is replaced by one of the objects that we use on this screen? 

![screenshot_625](https://user-images.githubusercontent.com/3046552/40062452-7e8b4042-5818-11e8-9b07-2944051f3380.png)

**Comment by kathy-phet** [comment]
I like it. K

Sent from my iPhone

On May 15, 2018, at 8:18 AM, Chris Malley <notifications@github.com<mailto:notifications@github.com>> wrote:


Perhaps something like this for the screen icon, where the apple is replaced by one of the objects that we use on this screen?

[screenshot_625]<https://user-images.githubusercontent.com/3046552/40062452-7e8b4042-5818-11e8-9b07-2944051f3380.png>

**Comment by pixelzoom** [comment]
Raising priority of this issue to "top". I need to know immediately if there are any changes/requirements that will impact Equality Explorer.

**Comment by amanda-phet** [comment]
One change is to use the square and circle from the Explore screen as the objects, as well as a third new object. Since you made these in code, I'm attaching a mockup below for reference, but I don't expect the triangle will look *exactly* like this. 
![image](https://user-images.githubusercontent.com/7852916/40439981-d1797438-5e79-11e8-8680-b3c65709133e.png)

As for the screen icon, I like what you did quite a bit, so we can use one of these objects for the icon. I'm fine with either the square or the triangle.

**Comment by pixelzoom** [comment]
Shapes were not made in code, they are in the .ai files.  I need both the shapes and their shadows.  Please add the triangle to these files:

https://github.com/phetsims/equality-explorer/blob/master/assets/basics-artwork-without-shadows.ai
https://github.com/phetsims/equality-explorer/blob/master/assets/basics-artwork-shadows.ai 

You can commit the changes yourself, or email me the .ai files, your choice.  But please use the most recent files in GitHub.

**Comment by pixelzoom** [comment]
> As for the screen icon, I like what you did quite a bit, so we can use one of these objects for the icon. I'm fine with either the square or the triangle.

I chose the square, since diagonal lines tend to look aliased on some platforms (particularly Firefox).  Here's a screenshot showing the Lab screen icon:

![screenshot_640](https://user-images.githubusercontent.com/3046552/40443494-bc0c5e8a-5e83-11e8-85f3-8597d4edeb44.png)

**Comment by amanda-phet** [comment]
New artwork is in the ai files and just committed the change. 

Icon: I'd like to make a few tweaks. 
(1) the background color should match the screen background. I'm fine changing the background on the screen (although I don't think we need to), but regardless the icon background should match.
(2) Can we add an equal sign? I prefer the original idea with an equal sign between the object and the picker. 
![image](https://user-images.githubusercontent.com/7852916/40452620-b16a6316-5e9f-11e8-8b94-7e13853df87d.png)

**Comment by pixelzoom** [comment]
Background color was an error and was fixed in a commit earlier today. It affected only the icon.

I will add the equals sign (that was an oversight.)

**Comment by pixelzoom** [comment]
@amanda-phet All requests are implemented in [1.0.0-dev.6](https://phet-dev.colorado.edu/html/equality-explorer-basics/1.0.0-dev.6/phet/equality-explorer-basics_en_phet.html). Please review.  If everything looks OK, feel free to close this issue.  If anything else occurs to you, let me know, happy to tweak.

**Comment by amanda-phet** [comment]
As discussed in status a few minutes ago, we'd like to make the picker ranges [0,20].

**Comment by pixelzoom** [comment]
@amanda-phet picker ranges are now [0,20], see master.

**Comment by pixelzoom** [comment]
@amanda-phet The default "weight" that bottoms out the scale is 30.  Now that the max value is 20, the scale can be bottomed out with only 2 objects.  Do you want to adjust the weight at which the scale bottoms out, as we did for the Animals scene (50)?

**Comment by amanda-phet** [comment]
>Do you want to adjust the weight at which the scale bottoms out, as we did for the Animals scene (50)?

That's a good idea, thanks.

One more change I forgot to mention: can we change the initial values to be 1, 2, and 3 for the circle, square, and triangle?

**Comment by pixelzoom** [comment]
What value do you want the scale to bottom out at?

> change the initial values to be 1, 2, and 3

Done.

**Comment by pixelzoom** [comment]
Slack discussion about `maxWeight` option for the scale is below.  We decided to leave it at the default, `maxWeight: 30`.

--
Amanda McGarry [11:27 AM]
I don’t think bottoming out at something like 50 will work because objects with a value of 1 will barely move the balance

on the Variables screen, didn’t we keep that bottom out value smaller?

Chris Malley [11:28 AM]
checking…

Amanda McGarry [11:29 AM]
We chose a max of 20 for the same reason we made pickers go to 20 in Proportion Playground.. there are a lot more “interesting” combinations. However, if it doesn’t work well, we can stick to 10

Chris Malley [11:29 AM]
All scenes bottom out at 30, with the exception of Animals which bottoms out at 50.

Amanda McGarry [11:30 AM]
The Animals scene is unique because the “values” are quite large, but those actual values aren’t visible so users can never know that we made that kind of change. They never have a 1 to measure against.

Chris Malley [11:30 AM]
dog=11, cat=4, turtle=6
Your call.

Amanda McGarry [11:30 AM]
yes I know the values, I am just saying the user doesn’t know the change we made because they never see the values
they could just as easily be 5.5, 2, and 3

Amanda McGarry [11:31 AM]
Anyway, I think the fact that a value of 1 can exist is the problem… so I don’t think we should change the bottom out value. but… going to 20 could be an issue so I’ll play around with it

Chris Malley [11:32 AM]
OK, I’ll leave it at 30.  Just wanted you to be aware of the consideration, since we changed the max range.

Amanda McGarry [11:32 AM]
I appreciate that, and at first I was on board but then I thought about it some more.
Yes, that is fine. I just didn’t want a back-and-forth within the already long github issue
 
I played a bit and am not bothered by the scale bottoming out with 2 heavy objects.

**Comment by pixelzoom** [comment]
@amanda-phet Anything else, or can we close this issue?

**Comment by amanda-phet** [comment]
Let's close it. If we make tweaks to the values box I can open a new issue.

**Comment by kathy-phet** [comment]
Would this bottom out weight only be on the lab screen? I think I favor keeping the original dynamics here.
K

Sent from my iPhone

On May 24, 2018, at 10:56 AM, Amanda McGarry <notifications@github.com<mailto:notifications@github.com>> wrote:


Do you want to adjust the weight at which the scale bottoms out, as we did for the Animals scene (50)?

That's a good idea, thanks.

One more change I forgot to mention: can we change the initial values to be 1, 2, and 3 for the circle, square, and triangle?

**Comment by pixelzoom** [comment]
So yes, it would affect only the Lab screen. The bottom-out weight is configurable for each scene, with a default of 30. The Animals scene in the _Basics_ screen currently is the only scene that overrides the default, with a value of 50.   And we've decided to keep the default for the Lab screen.  So it sounds like we're all on the same page.
